src/util/pelz_io.c

```diff
@@ -374,17 +391,107 @@ int read_pipe(char *msg)
       {
       case 'c':
         len = strcspn(msg, "\n");
-        path = new_charbuf(len - 11); //the number 11 is used because it the number of chars in "pelz -l -c "
-        memcpy(path.chars, &msg[11], (path.len));
-        free_charbuf(&path);
-        pelz_log(LOG_INFO, "Load cert call not added");
+        path = (char *) malloc((len - 10) * sizeof(char));  //the number 11 is used because it the number of chars in "pelz -l -c "
+        memcpy(path, &msg[11], len);
+
+        if (read_bytes_from_file(path, &data, &data_length))
+        {
+          pelz_log(LOG_ERR, "Unable to read file %s ... exiting", path);
+          free(path);
+          return 1;
+        }
+
+        pelz_log(LOG_DEBUG, "Read bytes from file %s", path);
+        memcpy(path_ext, &path[len - 15], 4); //4 is the set length of .nkl and .ski and 15 is the 11 deducted along with the 4 from the ext
```

Comment:
Probably cleaner to use strtok_r to find the extension, then set a variable that indicates the extension rather than copying out the bytes.

Reply:
Ignore, this comment is just wrong.